Sources/SwiftParser/Lexer/UnicodeScalarExtensions.swift

```diff
@@ -93,6 +93,31 @@ extension Unicode.Scalar {
     return true
   }
 
+  var isForbiddenRawIdentifierWhitespace: Bool {
+    let c = self.value
+    // This is the set of code points satisfying the `White_Space` property,
+    // excluding the set satisfying the `Pattern_White_Space` property, and
+    // excluding any other ASCII non-printables and Unicode separators. In
+    // other words, the only whitespace code points allowed in a raw
+    // identifier are U+0020, and U+200E/200F (LTR/RTL marks).
+    return (c >= 0x0009 && c <= 0x000D) as Bool
```

Comment:
Would it be better to express this as `isWhitespace && !isPermittedRawIdentifierWhitespace`?

Reply:
The parser explicitly avoids the use of `Unicode.Scalar.Properties` because they're not necessarily stable between Unicode versions and may change depending on the Unicode tables that are built into `libswiftCore`, so we don't want the behavior of the parser to change depending on things like what version of the operating system the compiler is running on.

It's probably _exceedingly rare_ that new whitespace will be added (or worse, that a non-whitespace code unit would become whitespace or vice versa), but hardcoding the code points means the behavior is deterministic and we can choose to make breaks via new language modes, if needed in the future.